### ISSUE TITLE
Warn about v-model components not handled with fireEvent.change

### ISSUE THREAD

**Issue by afontcu** [testing-library/vue-testing-library#83]
More often than not people relies on `fireEvent.input` or `fireEvent.change` to handle v-model powered inputs, thus running into issues with input value not being properly updated.

I'm wondering if we should provide some kind of warning when that happens.

Does it make sense?

**Comment by chiptus** [comment]
Do we have docs about it? I'm currently struggling with updating a custom component in a form using `fireEvent.update`

**Comment by afontcu** [comment]
We do have a section in the [API docs](https://testing-library.com/docs/vue-testing-library/api#updateelem-value), and some examples in the repo: 

https://github.com/testing-library/vue-testing-library/blob/993c87f8742fcaaaa942c8ae04b63f503f4e5824/src/__tests__/select.js
https://github.com/testing-library/vue-testing-library/blob/993c87f8742fcaaaa942c8ae04b63f503f4e5824/src/__tests__/form.js

**Comment by chiptus** [comment]
is `fireEvent.update` working with custom elements? like if I have a custom select box? in your examples you find the select component, but in the end, it's just a regular select. I'm trying to understand if all fireEvent.update expect is a component which can be used with v-model (because in my examples, it doesn't work)

**Comment by afontcu** [comment]
If that's the case, would you mind opening up a new issue with a reproduction link? All `fireEvent.update` [does](https://github.com/testing-library/vue-testing-library/blob/master/src/vue-testing-library.js#L128) is guessing the input type it was called upon, and update its value accordingly.

Thanks!! 🙌

**Comment by chiptus** [comment]
If I understand the code for [fireEvent](https://github.com/testing-library/dom-testing-library/blob/1233db0d343fa73fb2bc68fcbe816933adc2042e/src/events.js), this function just fires native events, right? if I want to update a component, I should update it as the user does, for example in my selector, I would click open, and choose the right value. right?

**Comment by afontcu** [comment]
> this function just fires native events
Yes, it is aligned with the guiding principle of Testing Library: keeping things as close as how your software is used.

> if I want to update a component, I should update it as the user does, for example in my selector, I would click open, and choose the right value. right?
Exactly! Interact with your component as any user would (or, well, as similar as possible).

---

`fireEvent.update` goes the extra mile and helps you with common use cases for `v-model`,  but I'm not 100% positive it would handle complex scenarios with custom events and compound components. If you find yourself stuck because of that, please share an example so we can work on it :)

**Comment by chiptus** [comment]
I actually found a very nice solution, I'll try to post an example

**Comment by ChamNouki** [comment]
I've got an issue that i think is related to this one.

I've got a custom input file component that use v-file-input from vuetify. My component emit `input` event to give selected file to the parent component through `v-model`. It works perfectly in browser but `emitted()` does not see any event.

**Comment by jameesjohn** [comment]
I noticed this while testing a component too.
I think having a warning of some sort will be great.

**Comment by jhack32** [comment]
@afontcu - Is this something that needs to be done? I can give it a try

**Comment by afontcu** [comment]
Hi @jhack32! Yeah sure! 🚀  I think it'd be great to add a warning message when using typing events on elements that we know may cause unexpected issues, and suggest people to use `update` instead :)

**Comment by JeromeDeLeon** [comment]
Should it trigger a warning for the input file?

**Comment by ChamNouki** [comment]
since issue occurred with input file too, it seems logic to me to trigger warning on input file too ?

**Comment by afontcu** [comment]
Hi! I'm not entirely sure what's the suggestion with input file – can't you simulate a file uploading using `fireEvent.update()`?

**Comment by afontcu** [comment]
Since this was released in #166, I'll go ahead and close this issue, feel free to open up a new one if we're missing some warnings!

thank you all!

**Comment by Serhansolo** [comment]
```ts
it('emits "filesSelected" when a file is selected via input change', async () => {
    const { container, emitted } = render(FileUpload, {
      props: { id: "test-file-input", labelText: "Upload File" },
    });
    const fileInput = container.querySelector('input[type="file"]') as HTMLInputElement;
    const files = [
      new File(["(⌐□_□)"], "chucknorris1.png", { type: "image/png" }),
      new File(["(□_□)"], "chucknorris2.png", { type: "image/jpeg" }),
    ];

    await fireEvent.change(fileInput, {
      target: {
        files,
      },
    });

    expect(emitted("filesSelected")).toBeTruthy();

    expect(emitted<[files: FileList]>("filesSelected")[0][0]).toHaveLength(2);
    expect(emitted<[files: FileList]>("filesSelected")[0][0][0]).toEqual(files[0]);

    expect(fileInput.value).toBe("");
  });
```

How do I solve this then? When I use update instead of change, I cannot add an array of files anymore...?



EDIT:

Nevermind, I found #179 this seems to work.
